server/src/Config.hs

```diff
@@ -79,6 +84,12 @@ configParser environment = Config
              help "Enable journaling. This only has an effect when periodic syncing is enabled.")
   where
     environ var = foldMap value (lookup var environment)
+
+    readFromEnvironment :: Read a => String -> Maybe a
+    readFromEnvironment var = case reads <$> lookup var environment of
```

Comment:
You can use [`Text.Read.readMaybe`](https://hackage.haskell.org/package/base-4.10.1.0/docs/Text-Read.html#v:readMaybe).